test/test-server/ccloud_router.go

```diff
@@ -9,63 +9,44 @@ import (
-	schemaRegistry      = "/api/schema_registries/{id}"
-	priceTable          = "/api/organizations/{id}/price_table"
-	paymentInfo         = "/api/organizations/{id}/payment_info"
-	promoCodeClaims     = "/api/organizations/{id}/promo_code_claims"
-	users               = "/api/users"
-	v2alphaAuthenticate = "/api/metadata/security/v2alpha1/authenticate"
-	signup              = "/api/signup"
-	launchDarklyProxy   = "/ldapi/sdk/eval/{env}/users/{user:[a-zA-Z0-9=\\-\\/]+}"
-	externalIdentities  = "/api/external_identities"
-	freeTrialInfo       = "/api/growth/v1/free-trial-info"
-)
+var ccloudHandlers = []route{
+	{"/api/accounts", handleEnvironments},
+	{"/api/accounts/{id}", handleEnvironment},
+	{"/api/env_metadata", handleEnvMetadata},
+	{"/api/external_identities", handleExternalIdentities},
+	{"/api/growth/v1/free-trial-info", handleFreeTrialInfo},
+	{"/api/login/realm", handleLoginRealm},
+	{"/api/metadata/security/v2alpha1/authenticate", handleV2Authenticate},
+	{"/api/organizations/{id}/payment_info", handlePaymentInfo},
+	{"/api/organizations/{id}/price_table", handlePriceTable},
+	{"/api/organizations/{id}/promo_code_claims", handlePromoCodeClaims},
+	{"/api/schema_registries/{id}", handleSchemaRegistry},
+	{"/api/service_accounts", handleServiceAccounts},
+	{"/api/service_accounts/{id}", handleServiceAccount},
+	{"/api/sessions", handleLogin},
+	{"/api/users", handleUsers},
+	{"/ldapi/sdk/eval/{env}/users/{user:[a-zA-Z0-9=\\-\\/]+}", handleLaunchDarkly},
+}
 
 type CloudRouter struct {
 	*mux.Router
-	srApiUrl   string
-	kafkaRPUrl string
+	srApiUrl string
 }
 
 // New CloudRouter with all cloud handlers
 func NewCloudRouter(t *testing.T, isAuditLogEnabled bool) *CloudRouter {
-	c := &CloudRouter{Router: mux.NewRouter()}
-	c.buildCcloudRouter(t, isAuditLogEnabled)
-	return c
-}
+	router := &CloudRouter{Router: mux.NewRouter()}
+	router.Use(defaultHeaderMiddleware)
 
-// Add handlers for cloud endpoints
-func (c *CloudRouter) buildCcloudRouter(t *testing.T, isAuditLogEnabled bool) {
-	c.HandleFunc(sessions, handleLogin(t))
-	c.HandleFunc(me, c.HandleMe(t, isAuditLogEnabled))
-	c.HandleFunc(loginRealm, handleLoginRealm(t))
-	c.HandleFunc(signup, c.HandleSignup(t))
-	c.HandleFunc(envMetadata, c.HandleEnvMetadata(t))
-	c.HandleFunc(launchDarklyProxy, c.HandleLaunchDarkly(t))
-	c.HandleFunc(externalIdentities, handleExternalIdentities(t))
-	c.addSchemaRegistryRoutes(t)
-	c.addEnvironmentRoutes(t)
-	c.addOrgRoutes(t)
-	c.addUserRoutes(t)
-	c.addV2AlphaRoutes(t)
-	c.addServiceAccountRoutes(t)
-	c.addGrowthRoutes(t)
-}
+	for _, route := range ccloudHandlers {
+		router.HandleFunc(route.path, route.handler(t))
+	}
 
-func (c CloudRouter) addV2AlphaRoutes(t *testing.T) {
-	c.HandleFunc(v2alphaAuthenticate, c.HandleV2Authenticate(t))
-	c.addRoutesAndReplies(t, v2Base, v2RoutesAndReplies)
+	router.HandleFunc("/api/me", handleMe(t, isAuditLogEnabled))
+	router.HandleFunc("/api/schema_registries", router.HandleSchemaRegistries(t))
```

Comment:
This one being the odd one out is unfortunate.

Reply:
Yeah, we can rewrite this later!